src/metrics/generated/proto/pipelinepb/pipeline.proto

```diff
@@ -80,8 +80,8 @@ message AppliedPipelineOp {
     ROLLUP = 2;
   }
   Type type = 1;
-  TransformationOp transformation = 2;
-  AppliedRollupOp rollup = 3;
+  TransformationOp transformation = 2 [(gogoproto.nullable) = false];
```

Comment:
what does this annotation do?

Reply:
it makes the generated field a value type, instead of a pointer type. a slice of value types can be reused. a slice of pointer types require allocating new structs for the slice.